static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -55,97 +111,127 @@ function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorP
+    value: team.id,
+    label: `#${team.slug}`,
+    leadingItems: <TeamAvatar team={team} size={18} />,
+  });
 
-  // Avatars/Badges in the Edit Selector Button
+  // Avatars/Badges in the Edit Access Selector Button
   const triggerAvatars =
-    selectedOptions.includes('_everyone') || !dashboardCreator ? (
+    selectedOptions.includes('_allUsers') || !dashboardCreator ? (
       <StyledBadge key="_all" text={'All'} />
+    ) : selectedOptions.length === 2 ? (
+      // Case where we display 1 Creator Avatar + 1 Team Avatar
+      <StyledAvatarList
+        key="avatar-list"
+        typeAvatars="users"
+        users={[dashboardCreator]}
+        teams={[teams.find(team => team.id === selectedOptions[1])!]}
+        maxVisibleAvatars={1}
+        avatarSize={25}
+      />
     ) : (
-      <StyledAvatarList key="avatar-list" users={[dashboardCreator]} avatarSize={25} />
+      // Case where we display 1 Creator Avatar + a Badge with no. of teams selected
+      <StyledAvatarList
+        key="avatar-list"
+        typeAvatars="users"
+        users={Array(selectedOptions.length).fill(dashboardCreator)}
+        maxVisibleAvatars={1}
+        avatarSize={25}
+      />
     );
 
-  const dropdownOptions = [
+  const allDropdownOptions = [
     makeCreatorOption(),
     {
-      value: '_everyone_section',
+      value: '_all_users_section',
       options: [
         {
-          value: '_everyone',
-          label: t('Everyone'),
-          disabled: dashboardCreator?.id !== currentUser.id,
+          value: '_allUsers',
+          label: t('All users'),
+          disabled: !isCurrentUserDashboardOwner,
         },
       ],
     },
-    // [WIP: Selective edit access to teams]
-    // {
-    //   value: '_teams',
-    //   label: t('Teams'),
-    //   options: teams.map(makeTeamOption),
-    //   showToggleAllButton: true,
-    //   disabled: true,
-    // },
+    {
+      value: '_teams',
+      label: t('Teams'),
+      options: teams.map(makeTeamOption),
+      showToggleAllButton: isCurrentUserDashboardOwner,
+      disabled: !isCurrentUserDashboardOwner,
+    },
   ];
 
-  // Handles state change when dropdown options are selected
-  const onSelectOptions = newSelectedOptions => {
-    const newSelectedValues = newSelectedOptions.map(
-      (option: {value: string}) => option.value
-    );
-    if (newSelectedValues.includes('_everyone')) {
-      setselectedOptions(['_everyone', '_creator', ...teamIds]);
-    } else if (!newSelectedValues.includes('_everyone')) {
-      setselectedOptions(['_creator']);
-    }
-  };
-
-  // Creates or modifies permissions object based on the options selected
-  function getDashboardPermissions() {
-    return {
-      isCreatorOnlyEditable: !selectedOptions.includes('_everyone'),
-    };
-  }
+  // Save and Cancel Buttons
+  const dropdownFooterButtons = (
+    <FilterButtons>
+      <Button
+        size="sm"
+        onClick={() => {
+          setMenuOpen(false);
+        }}
```

Comment:
I expected clicking "Cancel" would reset my selected state, can we add reset functionality here? We can follow it up in another PR if you want, not a blocker.

Reply:
Just wanted to note that I noticed opening and closing the selector reset my state, but cancelling didn't. I think we should make these two behaviours consistent :pray: